JavaScript development/Typescript/TypeScript.md

```diff
@@ -0,0 +1,119 @@
+# TypeScript
+
+## TypeScript+React
+*For general info on using JSX with TypeScript checkout the guidelines in TypeScript's official documentation [JSX · TypeScript](https://www.typescriptlang.org/docs/handbook/jsx.html)*.
+
+### Component Declaration
+
+When declaring a React class component in TypeScript, set two generic types that describe props and state.
+
+``` typescript
+export class ComponentName extends React.Component<{/*prop types*/}, {/*state types*/}> {
+    // ...
+}
+```
+
+Declare functional components as arrow functions
+
+``` typescript
+export const FunctionalComponent = (/*props*/): JSX.Element => {
```

Comment:
Is the return type valid in all cases? `null` is a valid use case (when we decide not to render a component), but I'm not sure it's assignable to `JSX.Element`.

Reply:
removed func. components